operators/pkg/controller/elasticsearch/version/zen2/compatibility.go

```diff
@@ -21,8 +24,25 @@ func IsCompatibleWithZen2(statefulSet appsv1.StatefulSet) bool {
 	return sset.ESVersionMatch(statefulSet, zen2VersionMatch)
 }
 
-// AtLeastOneNodeCompatibleWithZen2 returns true if the given StatefulSetList contains
-// at least one StatefulSet compatible with zen2.
-func AtLeastOneNodeCompatibleWithZen2(statefulSets sset.StatefulSetList) bool {
-	return sset.AtLeastOneESVersionMatch(statefulSets, zen2VersionMatch)
+// AllMastersCompatibleWithZen2 returns true if all master nodes in the given cluster can use zen2 APIs.
+// During a v6 -> v7 rolling upgrade, we can only call zen2 APIs once the current master is using v7,
+// which would happen only if there is no more v6 master-eligible nodes in the cluster.
+func AllMastersCompatibleWithZen2(c k8s.Client, es v1alpha1.Elasticsearch) (bool, error) {
+	masters, err := sset.GetActualMastersForCluster(c, es)
+	if err != nil {
+		return false, err
+	}
+	if len(masters) == 0 {
+		return false, nil
+	}
+	for _, pod := range masters {
+		v, err := label.ExtractVersion(pod.Labels)
+		if err != nil {
+			return false, err
+		}
+		if !zen2VersionMatch(*v) {
```

Comment:
This name is a bit odd. You seem to follow a `IsCompatibleWith` convention everywhere else. I would have expected something like `versionCompatibleWithZen2` or `versionSupportsZen2`